docs/flow/config-language/components.md

```diff
@@ -6,3 +6,157 @@ weight: 300
 ---
 
 # Components
```

Comment:
Overall, I would prefer that this document talk solely about components from the perspective of River: 

1. How does a user configure components? 
2. What are arguments? What are exports? 
3. How does a user reference a component? 

Going into this page, readers should already be familiar with the concepts of what components are and how the component controller works; we can link them to the other pages as a reminder without having to reexplain those ideas.

This is the one page of the River docs where I think we _can_ mention some Flow-specific things, since this is the only part of River that Flow directly impacts. Everything else should probably avoid saying things which are only true for Flow.

Reply:
I've added the relevant headers but I'm not sure if we should add more content to each.

Reply:
Should we talk about "scope" in one of these documents? Also, should we talk about the config endpoint and how we're re-creating the config file to be shown back to the user?

Reply:
> Should we talk about "scope" in one of these documents? 

I think the "scope" is probably still evolving so we might not want to be too detailed, but it does makes sense to say in the components page that only components can reference other components.

> Also, should we talk about the config endpoint and how we're re-creating the config file to be shown back to the user?

Maybe not right now, that config endpoint might go away in favor of the UI. (Though how the endpoint works feels like an implementation detail that the user shuoldn't need to know about)

Reply:
> but it does makes sense to say in the components page that only components can reference other components.

👍 I've added this under the expressions page that talks about referencing exports.